whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotObjectSynchronizer.java

```diff
@@ -31,31 +34,33 @@ public class SnapshotObjectSynchronizer {
     private static final int BATCH_SIZE = 10;
 
     private final Dummifier dummifierNrtm;
-    private final NrtmVersionInfoRepository nrtmVersionInfoRepository;
     private final SnapshotObjectRepository snapshotObjectRepository;
+    private final SourceRepository sourceRepository;
     private final WhoisObjectRepository whoisObjectRepository;
 
     SnapshotObjectSynchronizer(
         final Dummifier dummifierNrtm,
-        final NrtmVersionInfoRepository nrtmVersionInfoRepository,
         final SnapshotObjectRepository snapshotObjectRepository,
+        final SourceRepository sourceRepository,
         final WhoisObjectRepository whoisObjectRepository
     ) {
         this.dummifierNrtm = dummifierNrtm;
-        this.nrtmVersionInfoRepository = nrtmVersionInfoRepository;
         this.snapshotObjectRepository = snapshotObjectRepository;
+        this.sourceRepository = sourceRepository;
         this.whoisObjectRepository = whoisObjectRepository;
     }
 
-    NrtmVersionInfo initializeSnapshotObjects(final NrtmSource source) {
-        final String method = "initializeSnapshotObjects";
+    InitialSnapshotState initializeSnapshotObjects() {
+        LOGGER.info("initializeSnapshotObjects entered");
+        final Map<CIString, NrtmSourceModel> sourceMap = new HashMap<>();
+        for (final NrtmSource source : NrtmSourceHolder.getAllSources()) {
+            final NrtmSourceModel sourceModel = sourceRepository.createSource(source);
+            sourceMap.put(CIString.ciString(source.name()), sourceModel);
```

Comment:
NrtmSourceHolder is a component. I think it should deal with all logic related to source. What you think if that class has a function createAllSources? And NrtmSourceHolder should also have function to give us id based on source instead of creating a local map here? This will simplify this class to deal with only snapshotObject logic not nrtmSource? Also may be rename NrtmSourceHolder to NrtmSourceProcessor or something similar?

Reply:
done